src/hoc/withErrorHandler/withErrorHandler.js

```diff
@@ -5,20 +5,28 @@ import Aux from '../Aux/Aux';
 
 const withErrorHandler = ( WrappedComponent, axios ) => {
     return class extends Component {
+        interceptor1 = null
+        interceptor2 = null
+
         state = {
             error: null
         }
 
         componentWillMount () {
-            axios.interceptors.request.use(req => {
+            this.interceptor1 = axios.interceptors.request.use(req => {
                 this.setState({error: null});
                 return req;
             });
-            axios.interceptors.response.use(res => res, error => {
+            this.interceptor2 = axios.interceptors.response.use(res => res, error => {
                 this.setState({error: error});
             });
         }
 
+        componentWillUnmount() {
+            axios.interceptors.request.eject(this.interceptor1);
+            axios.interceptors.request.eject(this.interceptor2);
```

Comment:
the interceptor2 is in response, so this line should be: axios.interceptors.response.eject(this.interceptor2);